### PR TITLE
Fix _get_metadata_class to work recursively

### PR DESCRIPTION
**Pull request by wRAR**
A straightforward fix for #49 (at least assuming I've implemented BFS correctly), but I don't have a clear picture on what do `__orig_bases__` and `__origin__` do yet. Note that `__mro__` isn't suitable as it's depth-first.

### CODE REVIEW

**Review by Gallaecio — APPROVED** [review]

File: zyte_common_items/pages.py [hunk 1]
```diff
@@ -44,18 +45,24 @@ class HasMetadata(Generic[MetadataT]):
...
-        if not origin:
-            continue
-        if origin != HasMetadata:
-            continue
-        return get_args(base)[0]
-    return None
+        return _get_metadata_class(type(self))
+
+
+def _get_metadata_class(cls: type) -> Type[MetadataT]:
```
Comment (kmike) [inline zyte_common_items/pages.py:51]: Could you please a few more tests, for this function specifically, cvering the scenarios you considered while writing it?

Comment (kmike) [reply zyte_common_items/pages.py:51]: I also wonder if HasMetadata could be made an argument, though no strong opinion

**Comment by wRAR** [comment]
Requires new web-poet (https://github.com/scrapinghub/web-poet/pull/184)

File: setup.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
     install_requires=[
         "attrs>=21.3.0",
         "itemadapter>=0.8.0",
-        "web-poet>=0.7.0",
+        "web-poet>=0.14.0",
```
Comment (kmike) [inline setup.py:28]: I think "min" environment in tox.ini needs to be updated according to this change. Would it make sense to add a comment to setup.py, to ensure we don't forget to do it with other version updates in install_requires?

**Pull request merged**